pkg/cloud/cloud.go

```diff
@@ -835,6 +836,57 @@ func (c *cloud) getVolume(ctx context.Context, request *ec2.DescribeVolumesInput
 	return volumes[0], nil
 }
 
+func (c *cloud) GetVolumeAttachLimit(ctx context.Context, nodeID string) (int, error) {
+	instance, err := c.getInstance(ctx, nodeID)
```

Comment:
this means every node needs describeinstances permission right? I don't think we explicitly grant these in our e2e test cases so i'm not sure how the tests are passing (I will check) . In any case it would mean additional requirement for node policy and api call load so needs to be documented . If the same info is available in instance metadata it should attempt to retrieve from there first

Reply:
Metadata does not give enough information about the Network interfaces and the volumes attached. Hence Node needs describeInstances permission to look into the number of attachments.

Reply:
Then I think there needs to be a fallback in case the permission is not granted. Even if  we return the default limit without accounting for existing # of ENI / volumes, it is better than returning 0 jsut because the customer has not granted permission. Asking them for this permission is a breaking change and there are security reasons they will not want their nodes to be able to hit the aws api

Reply:
what about block-device-mapping/ebsN ? "The virtual devices associated with any Amazon EBS volumes. Amazon EBS volumes are only available in metadata if they were present at launch time or when the instance was last started. The N indicates the index of the Amazon EBS volume (such as ebs1 or ebs2). "

or network/interfaces/macs/mac/device-number for ENIS? https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-categories.html